src/Deferred.php

```diff
@@ -5,58 +5,16 @@
 namespace GraphQL;
 
 use GraphQL\Executor\Promise\Adapter\SyncPromise;
-use SplQueue;
-use Throwable;
 
-class Deferred
+class Deferred extends SyncPromise
 {
-    /** @var SplQueue|null */
-    private static $queue;
-
-    /** @var callable */
-    private $callback;
-
-    /** @var SyncPromise */
-    public $promise;
-
-    public function __construct(callable $callback)
-    {
-        $this->callback = $callback;
-        $this->promise  = new SyncPromise();
-        self::getQueue()->enqueue($this);
-    }
-
-    public static function getQueue() : SplQueue
-    {
-        if (self::$queue === null) {
-            self::$queue = new SplQueue();
-        }
-
-        return self::$queue;
-    }
-
-    public static function runQueue() : void
-    {
-        $queue = self::getQueue();
-        while (! $queue->isEmpty()) {
-            /** @var self $dequeuedNodeValue */
-            $dequeuedNodeValue = $queue->dequeue();
-            $dequeuedNodeValue->run();
-        }
-    }
-
-    public function then($onFulfilled = null, $onRejected = null)
+    public static function create(callable $executor) : self
```

Comment:
Why is this function added in addition to the constructor?

Reply:
As for me, it helps with fluent call chaining readability.

`(new A())->doSth()`  
vs
`A::create()->doSth()`

Not sure it was the reason here tho but I like it.

Reply:
Yes, that's the only reason